### PR TITLE
First draft of the journal entry specification for state changes

### PR DESCRIPTION
**Pull request by vpodzime**
Open for comments, tips, complaints and review.

### CODE REVIEW

File: spec.rst [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+  or some other similar value. **If some kind of such an identifier exists, it
+  must be specified.** It is only optional for devices that don't have any such
+  identifier.
+
+STATE
+  The new state of the device. If possible, the value should be one of the
+  following well-known states: ``discovered``, ``initialized``, ``failing``,
+  ``degraded``, ``failed``, ``missing``. If none of those are applicable, the
+  value should describe the new state of the device, preferably with a single
+  word.
```
Comment (dwlehman) [inline spec.rst]: A brief description of what each state represents would be useful.

File: spec.rst [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+
+   If the journal entry comes from the kernel, the *_KERNEL_* prefix is added
+   to some of the above fields, namely: *DEVICE*, *DEVICE_ID*, *SOURCE*,
+   *SOURCE_MAN*, *DETAILS*, *PRIORITY_DESC* due to how journald handles
+   structured logging from kernel.
+
+   Moreover, the *_KERNEL_DEVICE* field is very different, because kernel uses
+   a special form of the device identifier of the form ``+SUBSYSTEM:SYSNAME``,
+   for example ``+scsi:0:0:0:0`` for the first SATA/SCSI disk in the system
+   which is usually known as ``sda``.
```
Comment (dwlehman) [inline spec.rst:108]: Is there any tool or symlink somewhere like under `/dev/disk/` or `/sys` that would help resolve `+scsi:0:0:0:0` to `sda`?

File: spec.rst [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+PRIORITY
+  An integer priority value between ``0`` (emergency) and ``7`` (debug)
+  formatted as a decimal string. This field is compatible with syslog's priority
+  concept and describes the importance of the state change.
+
+PRIORITY_DESC
+  A one-word description of *PRIORITY*. Possible values are: ``emergency``,
+  ``alert``, ``critical``, ``error``, ``warning``, ``notice``, ``info``,
+  ``debug``.
+
```
Comment (tasleson) [inline spec.rst:93]: Is there a reason we need both PRIORITY and PRIORITY_DESC?  You can derive one from the other..

File: spec.rst (hunk 2 above)
Comment (vpodzime) [reply spec.rst:108]: Yes, there's the following path ``/sys/bus/scsi/devices/0\:0\:0\:0/block/sda`` and for the other direction, there's the path ``/sys/class/block/sda/device/scsi_device/0\:0\:0\:0/``.

File: spec.rst (hunk 3 above)
Comment (vpodzime) [reply spec.rst:93]: ``PRIORITY_DESC`` is really only for ``journactl -x`` to show a nice human-readable priority description in the catalog entry/template. See http://www-rhstorage.rhcloud.com/blog/vpodzime/reporting-and-monitoring-storage-events#human-readability for what exactly I mean. And ``PRIORITY`` is a standard field [recommended/required by *journald*](https://www.freedesktop.org/software/systemd/man/systemd.journal-fields.html)

**Pull request merged**